common/strings/diff.cc

```diff
@@ -44,16 +44,16 @@ static char EditOperationToLineMarker(Operation op) {
 LineDiffs::LineDiffs(absl::string_view before, absl::string_view after)
     : before_text(before),
       after_text(after),
-      before_lines(SplitLines(before_text)),
-      after_lines(SplitLines(after_text)),
+      before_lines(SplitLinesWithLineTerminator(before_text)),
```

Comment:
I did this change when implementing autofix for the "posix-eof" rule. When all `\n` characters are stripped, the diff algorithm is unable to detect that `\n` has been added. While this can really happen only in the last line of a file, it is easier to just keep all newlines.